beacon_node/eth2-libp2p/tests/rpc_tests.rs

```diff
@@ -106,20 +107,19 @@ fn test_status_rpc() {
     });
 
     // execute the futures and check the result
-    let test_result = Arc::new(Mutex::new(false));
+    let test_result = Arc::new(AtomicBool::new(false));
     let error_result = test_result.clone();
     let thread_result = test_result.clone();
     tokio::run(
         sender_future
             .select(receiver_future)
             .timeout(Duration::from_millis(1000))
-            .map_err(move |_| *error_result.lock().unwrap() = false)
+            .map_err(move |_| error_result.store(false, Relaxed))
```

Comment:
AtomicBool was recommended by Clippy. I think Relaxed should be enough here but I may be wrong. It passes the tests on my computer. Can someone confirm that this piece of code is ok? :)

Reply:
Yep Relaxed is fine. We are not reading any other variables and just use this as a flag to end the test